main.cpp

```diff
@@ -103,22 +130,35 @@ struct HeapA
     }
 };
 
-#include <iostream>
-#include <cmath>
-#include <functional>
-#include <memory>
-
 
 template <typename NumericType>
 struct Numeric
 {
+    
     using Type = Temporary<NumericType>;
 
     Numeric(Type num) : numberPtr(std::make_unique<Type>(num)){}
 
-    ~Numeric()
+//Rule of 3
+    Numeric(NumericType num) : numberPtr(std::make_unique<Type>(num)){}
+
+    template<typename OtherType>
+    Numeric& operator= (const OtherType& o)
     {
-        numberPtr = nullptr;
+        *numberPtr = static_cast<NumericType>(o); 
+        return *this; 
+    }
```

Comment:
This constructor is not part of the Rule of 3, because it's constructor argument is not a `const Numeric&`
This assignment operator is not part of the Rule of 3, because it is not `operator=(const Numeric&)`

`Numeric(NumericType)` is a converting constructor.
`operator=(const OtherType&)` is an overloaded assignment operator.
- [ ] ask questions in Slack if needed.